hamocc/mo_control_bgc.F90

```diff
@@ -16,86 +16,151 @@
 ! You should have received a copy of the GNU Lesser General Public License 
 ! along with BLOM. If not, see https://www.gnu.org/licenses/.
 
+MODULE mo_control_bgc
+  !***********************************************************************
+  !
+  !**** *MODULE mo_control_bgc* - control variables for bgc modules.
+  !
+  !     S.Legutke,        *MPI-MaD, HH*    28.02.02
+  !
+  !     Modified
+  !     --------
+  !     J.Schwinger,      *Uni Research, Bergen*   2018-04-12
+  !     - removed unused variables
+  !     
+  !     Purpose
+  !     -------
+  !     - declaration
+  !
+  !
+  !**********************************************************************
+  implicit none
 
-      MODULE mo_control_bgc
-!***********************************************************************
-!
-!**** *MODULE mo_control_bgc* - control variables for bgc modules.
-!
-!     S.Legutke,        *MPI-MaD, HH*    28.02.02
-!
-!     Modified
-!     --------
-!     J.Schwinger,      *Uni Research, Bergen*   2018-04-12
-!     - removed unused variables
-!     
-!     Purpose
-!     -------
-!     - declaration
-!
-!
-!**********************************************************************
-      implicit none
+  ! Logical unit number for I/O.
+  INTEGER :: io_stdo_bgc        !  standard out.
+
+  ! File containing namelists
+  CHARACTER(LEN=:), ALLOCATABLE, PROTECTED :: bgc_namelist
 
-! Logical unit number for I/O.
-      INTEGER, save :: io_stdo_bgc        !  standard out.
+  ! Control variables
+  REAL    :: dtbgc                    !  time step length [sec].
+  REAL    :: dtb                      !  time step length [days].
+  INTEGER :: ndtdaybgc                !  time steps per day.
 
-! File containing namelists
-      CHARACTER(LEN=:), ALLOCATABLE, PROTECTED :: bgc_namelist
+  INTEGER :: ldtbgc                   !  time step number from bgc restart file
+  INTEGER :: ldtrunbgc                !  actual time steps of run.
 
-! Control variables
-      REAL,    save :: dtbgc              !  time step length [sec].
-      REAL,    save :: dtb                !  time step length [days].
-      INTEGER, save :: ndtdaybgc          !  time steps per day.
```

Comment:
I think it is actually correct to have save attributes here. Unless a module variable is used in the main program it can go out of scope and the Fortran standard doesn't require that the variable's value is retained (as far as I know).

Reply:
@JorgSchwinger - for module scope my understanding is that save is now on by default.

Reply:
> @JorgSchwinger - for module scope my understanding is that save is now on by default.

This is true. Note that this is a change from the behavior in Fortran 2003 and before. Beginning with Fortran 2008 (and universally true in contemporary compilers to my knowledge), all module-scope variables have the save attribute by default.

Reply:
@JorgSchwinger - CESM has been leverage this to remove save attributes from module variables in new development versions of the code.

Reply:
Ok, thanks for clarifying this - I see my Fortran knowledge is becoming outdated...